core-feature-pack/src/main/resources/content/bin/common.ps1

```diff
@@ -0,0 +1,256 @@
+			echo "ERROR: The use of -Djava.security.manager has been removed. Please use the -secmgr command line argument or SECMGR=true environment variable."
+			exit
+		}elseif ($arg -eq '-secmgr'){
+			$global:SECMGR = $true
+		}elseif ($arg -eq '--background'){
+			$global:RUN_IN_BACKGROUND = $true
+		}else{
+			$res+=$arg
+		}
+	}
+	return $res
+}
+
+Function Start-WildFly-Process {
+ Param(
+   [Parameter(Mandatory=$true)]
+   [string[]] $programArguments,
+   [boolean] $runInBackground = $false   
+
+) #end param
+
+	if(($JBOSS_PIDFILE -ne '') -and (Test-Path $JBOSS_PIDFILE)) {
+		$processId = gc $JBOSS_PIDFILE
+		if ($processId -ne $null){
+			$proc = Get-Process -Id $processId -ErrorAction SilentlyContinue
+		}
+		if ($proc -ne $null){
+			Write-Warning "Looks like a server process is already running. If it isn't then, remove the $JBOSS_PIDFILE and try again"
+			return
+		}else{
+			Remove-Item $JBOSS_PIDFILE
+		}
+	}
+
+	if($runInBackground) {	
+		$process = Start-Process -FilePath $JAVA -ArgumentList $programArguments -NoNewWindow -RedirectStandardOutput $global:CONSOLE_LOG -WorkingDirectory $JBOSS_HOME -PassThru
+		$processId = $process.Id;
+		echo "Started process in background, process id: $processId" 
+		if ($JBOSS_PIDFILE -ne $null){
+			$processId >> $JBOSS_PIDFILE  	
+		}
+	} else {
+		try{
+			pushd $JBOSS_HOME
+			& $JAVA $programArguments
+		}finally{
+			pushd $SCRIPTS_HOME
+		}
+	}
+
+}
+
+Function Set-Global-Variables {
+PARAM(
+[Parameter(Mandatory=$true)]
+   [string]$baseDir
+)
+
+	# determine the default base dir, if not set
+	$global:JBOSS_BASE_DIR = $baseDir;
+	
+	# determine the default log dir, if not set
+	$global:JBOSS_LOG_DIR = Get-Env JBOSS_LOG_DIR $JBOSS_BASE_DIR\log
+	
+	# determine the default configuration dir, if not set
+	$global:JBOSS_CONFIG_DIR = Get-Env JBOSS_CONFIG_DIR $JBOSS_BASE_DIR\configuration
+	
+	$global:CONSOLE_LOG = $JBOSS_LOG_DIR + '\console.log'
+}
+
+Function Set-Global-Variables-Standalone {
+	$dir = Get-Env JBOSS_BASE_DIR $JBOSS_HOME\standalone
+	Set-Global-Variables -baseDir $dir
+}
+
+Function Set-Global-Variables-Domain {
+	$dir = Get-Env JBOSS_BASE_DIR $JBOSS_HOME\domain
+	Set-Global-Variables -baseDir $dir	
+}
+
+
+# Setup JBOSS_HOME
+if( Test-Path env:JBOSS_HOME) {
+  $SANITIZED_JBOSS_HOME = (Get-Item env:JBOSS_HOME).FullName
+  if( $SANITIZED_JBOSS_HOME -ne $RESOLVED_JBOSS_HOME) {
+    echo "WARNING JBOSS_HOME may be pointing to a different installation - unpredictable results may occur."
+    echo ""
+  }
+  $JBOSS_HOME=$SANITIZED_JBOSS_HOME
+} else {
+    # get the full path (without any relative bits)
+    $JBOSS_HOME=$RESOLVED_JBOSS_HOME
+}
+
+# Setup the JVM
+if (!(Test-Path env:JAVA)) {
+  if( Test-Path env:JAVA_HOME) {
+    $JAVA = (Get-ChildItem env:JAVA_HOME).Value + "\bin\java"
+  } else {
+    $JAVA = 'java'
+  }
+}
+
+# determine the default module path, if not set
+$JBOSS_MODULEPATH = Get-Env JBOSS_MODULEPATH $JBOSS_HOME\modules
+
+Set-Global-Variables-Standalone
```

Comment:
this is function call, and yes it is somewhat counter intuitive, but what is does is by default sets the standalone variables, which are then overridden by call to Set-Global-Variables-Domain in domain.ps1 

by default all scrips behave as standalone ones, and set all variables for standalone server.
domain script than just overrides the stuff that is different for domain.

Basically try to do as much as possible in single place and only override/redefine stuff that is different.